peer.js

```diff
@@ -253,6 +254,8 @@ TChannelPeer.prototype.setScoreStrategy = function setScoreStrategy(ScoreStrateg
 TChannelPeer.prototype.invalidateScore = function invalidateScore(reason) {
     var self = this;
 
+    self._weightedRange = null;
```

Comment:
Originally the idea was to eagerly compute the range here everytime.

What you have instead is lazy caching.

The eager case might burn more cpu, but maybe during the less critical path(s) off of peer selection.